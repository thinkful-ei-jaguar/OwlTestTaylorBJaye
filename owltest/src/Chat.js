import React from 'react';

export default function Chat (props) {
    let date = new Date(props.timestamp).toTimeString();
    
    return (
        <div className="messages">
            <p>{props.participantId}</p>
            <p>{props.type}</p>
            <p>{date}</p>
        </div>
    )

}