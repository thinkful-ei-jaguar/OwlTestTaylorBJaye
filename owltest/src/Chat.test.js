import React from 'react';
import Chat from './Chat';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';

describe('Chat Component', () => {
    it('render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Chat
            participantId={2}
            type="thumbs-up"
            timestamp={1548852484247}
            />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders Chat UI as expected', () => {
        const flower = renderer
        .create(<Chat
            participantId={2}
            type="thumbs-up"
            timestamp={1548852484247}
            />)
        .toJSON();
        expect(flower).toMatchSnapshot();
    });
});
