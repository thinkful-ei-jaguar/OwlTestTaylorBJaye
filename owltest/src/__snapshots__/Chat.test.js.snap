// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Component renders Chat UI as expected 1`] = `
<div
  className="messages"
>
  <p>
    2
  </p>
  <p>
    thumbs-up
  </p>
  <p>
    07:48:04 GMT-0500 (Eastern Standard Time)
  </p>
</div>
`;
